test/range/detail/random_access_iterator_test.cpp

```diff
@@ -0,0 +1,472 @@
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+#include <seqan3/range/detail/random_access_iterator.hpp>
+
+#include <gtest/gtest.h>
+#include <sstream>
+#include <vector>
+
+class random_access_iterator_test_fixture: public ::testing::Test
+{
+protected:
+
+   std::vector<uint8_t> v_empty{};
+   const std::vector<uint8_t> v_const_empty{};
+   std::vector<uint8_t> *v, *v2, *v3, *v4, *w, *w2;
+   const std::vector<uint8_t> *v_const, *v2_const, *v3_const, *v4_const, *w_const, *w2_const;
+   std::array<long int, 3> a, b;
+   const std::array<long int, 3> a_const[3] = {11, 22, 33};
+
+   virtual void SetUp( ) {
+       // code here will execute just before the test ensues
+       v = new std::vector<uint8_t>                 {'a', 't'};
+       v_const = new const std::vector<uint8_t>     {'a', 't'};
+       v2 = new std::vector<uint8_t>                {'a', 'u'};
+       v2_const = new const std::vector<uint8_t>    {'a', 'u'};
+       v3 = new std::vector<uint8_t>                {'a', 't', 'z'};
+       v3_const = new const std::vector<uint8_t>    {'a', 't', 'z'};
+       v4 = new std::vector<uint8_t>                {'a', 'u', 'v', 'w', 'x'};
+       v4_const = new const std::vector<uint8_t>    {'a', 'u', 'v', 'w', 'x'};
+       w = new std::vector<uint8_t>                 {'c', 't'};
+       w_const = new const std::vector<uint8_t>     {'c', 't'};
+       w2 = new std::vector<uint8_t>                {'b', 'v'};
+       w2_const = new const std::vector<uint8_t>    {'b', 'v'};
+       a = {11, 22, 33};
+   }
+
+   virtual void TearDown( ) {
+       delete v, delete v2, delete v3, delete v4, delete w, delete w2;
+       delete v_const, delete v2_const, delete v3_const, delete v4_const, delete w_const, delete w2_const;
+   }
+
```

Comment:
I still think that the new and delete are not required here. For the const vectors, they are actually not changed, and for the non-const vectors you can just **assign** in the SetUp function, instead of re-allocating. Teardown should not be required.
Also please adhere to the style guide in this class (placement of braces and and parentheses)!